### ISSUE TITLE
'capture' func catch fatal error.

### ISSUE THREAD

**Issue by ftitreefly** [jakeheis/SwiftCLI#80]
`Fatal error: Couldn't parse data into string using Unicode (UTF-8): file /tmp/build/Course/checkouts/SwiftCLI/Sources/SwiftCLI/Stream.swift, line 213
2019-05-29 21:45:35.852414+0800 Course[2522:79439] Fatal error: Couldn't parse data into string using Unicode (UTF-8): file /tmp/build/Course/checkouts/SwiftCLI/Sources/SwiftCLI/Stream.swift, line 213
Program ended with exit code: 9`

```
    private static func captureCurl(args: [String], timeout: Int? = nil) throws -> String {
        return try capture("curl", arguments: args).rawStdout
    }
```

**Comment by ftitreefly** [comment]
Sometimes the error occurs when the stream reads more than 4K bytes.

**Comment by ftitreefly** [comment]
Perhaps, combine the **Bytes** first and convert them to **String** without making errors.

```
public class CaptureStream: ProcessingStream {
   
    // private var content = ""
    private var chunkData = Data()

    public init() {
        let pipe = Pipe()
        self.processObject = pipe
        self.writeHandle = pipe.fileHandleForWriting
        
        let readStream = ReadStream.for(fileHandle: pipe.fileHandleForReading)
        queue.async { [weak self] in
//            while let chunk = readStream.read() {
//                self?.content += chunk
//            }
            while let chunk = readStream.readData() {
                self?.chunkData += chunk
            }

            self?.semaphore.signal()
        }
    }

    public func readAll() -> String {
        if !waited {
            waitToFinishProcessing()
        }
        // return content
         return String(data: chunkData, encoding: encoding) ?? ""
    }
```

**Comment by jakeheis** [comment]
Thanks for the issue report! This is fixed in `5.3.0`. You can now avoid dealing with strings entirely:

```
private static func captureCurl(args: [String], timeout: Int? = nil) throws -> Data {
        return try capture("curl", arguments: args).stdoutData
}
```
